In [1]:
import os

BASE_DIR= "/Users/jeongahlee/Documents/github folder/CubiGraph5K/cubicasa5k-3493-cubigraph"
CATEGORIES = ["colorful", "high_quality", "high_quality_architectural"]


def count_subfolders_and_info(base_dir, category, total_num, info_num):
    cat_path = os.path.join(base_dir, category)
    if not os.path.isdir(cat_path):
        print(f"[{category}] 경로가 없습니다.")
        return

    subdirs = [
        name for name in os.listdir(cat_path)
        if os.path.isdir(os.path.join(cat_path, name))
    ]
    total = len(subdirs)

    info_count = 0
    for sub in subdirs:
        info_path = os.path.join(cat_path, sub, "info.txt")
        if os.path.isfile(info_path):
            info_count += 1
            info_num += 1

    print(f"[{category}] 하위 폴더 수: {total}, info.txt 있는 폴더 수: {info_count}")
    total_num += total
    return total_num, info_num

def main():
    total_num = 0
    info_num = 0
    for cat in CATEGORIES:
        total_num, info_num = count_subfolders_and_info(BASE_DIR, cat, total_num, info_num)
        print(total_num)
        print(info_num)

if __name__ == "__main__":
    main()

[colorful] 하위 폴더 수: 199, info.txt 있는 폴더 수: 199
199
199
[high_quality] 하위 폴더 수: 709, info.txt 있는 폴더 수: 709
908
908
[high_quality_architectural] 하위 폴더 수: 2585, info.txt 있는 폴더 수: 2585
3493
3493


In [1]:
import os

BASE_DIR = "../cubicasa5k-sample-69-with-hard"
CATEGORY_1 = ["easy", "medium"]
CATEGORIES = ["colorful", "high_quality", "high_quality_architectural"]

def count_subfolders_and_info(base_dir, level1, level2, total_num, info_num):
    cat_path = os.path.join(base_dir + "/" + level1, level2)
    if not os.path.isdir(cat_path):
        print("cat_path:", cat_path)
        print(f"[{level1}/{level2}] 경로가 없습니다.")
        return total_num, info_num

    subdirs = [
        name for name in os.listdir(cat_path)
        if os.path.isdir(os.path.join(cat_path, name))
    ]
    total = len(subdirs)

    info_count = 0
    for sub in subdirs:
        info_path = os.path.join(cat_path, sub, "info.txt")
        if os.path.isfile(info_path):
            info_count += 1
            info_num += 1

    print(f"[{level1}/{level2}] 하위 폴더 수: {total}, info.txt 있는 폴더 수: {info_count}")
    total_num += total
    return total_num, info_num

def main():
    total_num = 0
    info_num = 0
    for level1 in CATEGORY_1:
        for level2 in CATEGORIES:
            total_num, info_num = count_subfolders_and_info(BASE_DIR, level1, level2, total_num, info_num)

    print(f"\n총 하위 폴더 수: {total_num}")
    print(f"총 info.txt 포함된 폴더 수: {info_num}")

if __name__ == "__main__":
    main()


[easy/colorful] 하위 폴더 수: 3, info.txt 있는 폴더 수: 3
[easy/high_quality] 하위 폴더 수: 5, info.txt 있는 폴더 수: 5
[easy/high_quality_architectural] 하위 폴더 수: 11, info.txt 있는 폴더 수: 11
[medium/colorful] 하위 폴더 수: 13, info.txt 있는 폴더 수: 13
[medium/high_quality] 하위 폴더 수: 10, info.txt 있는 폴더 수: 10
[medium/high_quality_architectural] 하위 폴더 수: 17, info.txt 있는 폴더 수: 17

총 하위 폴더 수: 59
총 info.txt 포함된 폴더 수: 59


[colorful] 하위 폴더 수: 199, info.txt 있는 폴더 수: 199
199
[high_quality] 하위 폴더 수: 709, info.txt 있는 폴더 수: 709
908
[high_quality_architectural] 하위 폴더 수: 2585, info.txt 있는 폴더 수: 2585
3493